sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -502,11 +502,31 @@ private[sql] object DataSourceV2Strategy {
 
   private def translateLeafNodeFilterV2(predicate: Expression): Option[Predicate] = {
     predicate match {
-      case PushablePredicate(expr) => Some(expr)
+      case PushablePredicate(expr) =>
+        if (expr.children().length == 2) {
+          expr.children()(0) match {
+            case LiteralValue(_, _) =>
+              Some(new Predicate(flipComparisonFilterName(expr.name()),
```

Comment:
I wonder if we should be more strict and only flip the filter iff the operator is in the set `{ >, <, >=, <=}`